src/gilded_rose.js

```diff
@@ -13,50 +13,25 @@ items.push(new Item('Sulfuras, Hand of Ragnaros', 0, 80));
 items.push(new Item('Backstage passes to a TAFKAL80ETC concert', 15, 20));
 items.push(new Item('Conjured Mana Cake', 3, 6));
 
-function update_quality() {
-  for (var i = 0; i < items.length; i++) {
-    if (items[i].name != 'Aged Brie' && items[i].name != 'Backstage passes to a TAFKAL80ETC concert') {
-      if (items[i].quality > 0) {
-        if (items[i].name != 'Sulfuras, Hand of Ragnaros') {
-          items[i].quality = items[i].quality - 1
-        }
-      }
-    } else {
-      if (items[i].quality < 50) {
-        items[i].quality = items[i].quality + 1
-        if (items[i].name == 'Backstage passes to a TAFKAL80ETC concert') {
-          if (items[i].sell_in < 11) {
-            if (items[i].quality < 50) {
-              items[i].quality = items[i].quality + 1
-            }
-          }
-          if (items[i].sell_in < 6) {
-            if (items[i].quality < 50) {
-              items[i].quality = items[i].quality + 1
-            }
-          }
-        }
-      }
-    }
-    if (items[i].name != 'Sulfuras, Hand of Ragnaros') {
-      items[i].sell_in = items[i].sell_in - 1;
-    }
-    if (items[i].sell_in < 0) {
-      if (items[i].name != 'Aged Brie') {
-        if (items[i].name != 'Backstage passes to a TAFKAL80ETC concert') {
-          if (items[i].quality > 0) {
-            if (items[i].name != 'Sulfuras, Hand of Ragnaros') {
-              items[i].quality = items[i].quality - 1
-            }
-          }
-        } else {
-          items[i].quality = items[i].quality - items[i].quality
-        }
-      } else {
-        if (items[i].quality < 50) {
-          items[i].quality = items[i].quality + 1
-        }
-      }
-    }
-  }
+function normal_aging(x) {
+  x.quality -= 1
+  if (x.sell_in<=0) x.quality-=1
+  if (x.quality < 0) x.quality = 0
 }
+
+function update_quality() {items.forEach((x) => {
+    if (x.name!='Aged Brie' && x.name!='Sulfuras, Hand of Ragnaros' && x.name!='Backstage passes to a TAFKAL80ETC concert') {
```

Comment:
Mulitple and nested `if`'s is a confusing and rigid design approach. It is what made the original implementation hard to understand and modify. Continuing the same design approach will ultimately lead to another incomprehensible pile of `if`'s.

Refactoring is not amout keeping the same appraoch and making it a little more readable. It's about being able to toss out the original design and implement something totally new and more extensible (once you have some tests to make sure you don't break anything)

Do this exercise again, but force yourself to come to a design that **never** uses the item name to decide what to do and **never** uses `switch`, `else`, or `else if` - it is possible!